android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java

```diff
@@ -217,6 +217,14 @@ public void addPermissionObserver() {
       }
    }
 
+   @ReactMethod
```

Comment:
Can you make removeObservers() method call these new methods? Instead of repeating the logic

Reply:
are you saying something like
```   
public void removePermissionObserver() {
    if (hasSetPermissionObserver) {
        OneSignal.getPermissionStateChangesObserver().removeObserver(this);
        hasSetPermissionObserver = false;
    }
}
```
not too familiar with the android sdk but wouldn't that cause weirdness when the appContext is null?
https://github.com/OneSignal/OneSignal-Android-SDK/blob/68d6305143a7640f5be8e07a4a6bc7f1357046fa/OneSignalSDK/onesignal/src/main/java/com/onesignal/OneSignal.java#L2833-L2840

Reply:
@frankenthumbs ,
She's referring to [this method](https://github.com/OneSignal/react-native-onesignal/blob/cab98a795595fe0984ad81b6107202e6bde15f21/android/src/main/java/com/geektime/rnonesignalandroid/RNOneSignal.java#L122) where the underlying native SDK is called and the `hasSet*` booleans are flipped. Since this logic is already occurring in each of your new individual member methods, we can simply call them here instead.

Reply:
ahh gotcha.. re-reading jaesmine's comment, it totally makes sense... commit incoming!

Reply:
thanks!